receiver/k8sobjectsreceiver/receiver.go

```diff
@@ -48,22 +54,68 @@ func newReceiver(params receiver.CreateSettings, config *Config, consumer consum
 		return nil, err
 	}
 
-	return &k8sobjectsreceiver{
-		client:   client,
-		setting:  params,
-		consumer: consumer,
-		objects:  config.Objects,
-		obsrecv:  obsrecv,
-		mu:       sync.Mutex{},
-	}, nil
+	objReceiver := &k8sobjectsreceiver{
+		client:         client,
+		setting:        params,
+		logger:         params.Logger,
+		consumer:       consumer,
+		objects:        config.Objects,
+		obsrecv:        obsrecv,
+		mu:             sync.Mutex{},
+		leaderElection: config.LeaderElection,
+	}
+
+	if config.LeaderElection.Enabled {
+		objReceiver.leaderElectionClient, err = config.getClient()
+		if err != nil {
+			return nil, err
+		}
+
+		// use component "type-name" as resource lock name.
```

Comment:
Why this? Can't we have "component/name"? Is it a problem with slashes? If so, perhaps it's simpler to just run a strings.Replace(params.ID.String(), "/", "-") ?

Reply:
I would also split this into its function and write a couple of unit tests for it.

Reply:
yes, `/` is invalid in k8s reource name spec.

Reply:
> I would also split this into its function and write a couple of unit tests for it.

ok